cli/popper/commands/cmd_dot.py

```diff
@@ -0,0 +1,66 @@
+import os
+
+import click
+import hcl
+import popper.utils as pu
+from ..cli import pass_context
+import types
+
+
+@click.argument(
+    'wfile', required=True)
+@click.command('dot', short_help='Generates a dot file '
+                                 '[Used for graphical representations]')
+@pass_context
+def cli(ctx, wfile):
+    """Creates a dot file"""
+    if not os.path.isfile(wfile):
+        pu.fail("File {} not found.\n".format(wfile))
+        exit(1)
+
+    with open(wfile, 'r') as fp:
+        wf = hcl.load(fp)
+
+    graph = "digraph G {\n"
+    name = list(wf["workflow"].keys())[0]
+    pu.info("Dot file for %s workflow\n" % name)
+    pu.info("`ascii.dot` created\n\n")
+
+    action = wf["workflow"][name]["resolves"]
+    parent_action = cur_action = action
+
+    if not isinstance(cur_action, str):
+        cur_action = cur_action[0]
+
+    graph = add(parent_action, cur_action, wf["action"], graph) + "}\n"
+
+    with open('ascii.dot', 'w') as fp:
+        fp.write(graph)
+
+    os.system("awk '!seen[$0]++' ascii.dot > out")
```

Comment:
can you please explain why we need to do this? Is this due to the fact that one more than one action can depend on the same action? If that's the case, then maybe we can do this in python. This help to remove the dependency with `awk`, which in fact is also present [here](https://github.com/systemslab/popper/blob/master/cli/popper/scm.py#L61).

Reply:
Yes you are right, L40 was to curb duplicate statements because duplicate edges were generated due to multiple actions depending on the same action. Though it seems like as a part of fix, I forgot to integrate it properly. Thanks for pointing it out. 
I think I can remove the generation of duplicate edges using a visited array to keep track of actions visited. It will also remove the dependency on `awk`. 
( I thought that `awk` was fine as it was being used at other parts of code. I will keep the code pure python from now on)